### PR TITLE
Fix separate make and install execution error issue.

### PR DESCRIPTION
**Pull request by JackBoosY**
The install command has not been executed, it looks like it was caused by the separation of make and install.

Related: #8473.

### CODE REVIEW

**Comment by JackBoosY** [comment]
@Christsnatcher @spindensity @oracleoftroy @timautry 
Could you test it?
Thanks.

**Comment by spindensity** [comment]
I've tested it. It works perfectly.

Thanks.

**Comment by oracleoftroy** [comment]
To be honest, I'm a bit surprised it worked, but it worked for me as well. 

I don't know what vcpkg's policy is for making sure vcpkg still works if installed in paths with spaces, but I suspect the change on line 55 of `vcpkg_build_make.cmake` to remove the quotes around "${_VCPKG_PROJECT_SUBPATH}make" would break this use case. But if those quotes are contributing to this error in the first place, then I vote that they go.

**Comment by JackBoosY** [comment]
@oracleoftroy Well, I think this is due to the simultaneous use of `cmake_parse_arguments` and `execute_process`. I don't know what cmake does in the command for the arguments behind the command.

**Comment by oracleoftroy** [comment]
Hmm, I see. I wish I understood cmake and vcpkg internals better. I got stuck at this part. It just surprised me that a few quotes added and removed was all it took. :)

**Comment by Christsnatcher** [comment]
I unfortunately still get the same error as before.

**Comment by Christsnatcher** [comment]
Thank you @JackBoosY, working fine here either.

**Comment by Neumann-A** [comment]
@JackBoosY 
The difference in cmake is:
` "i am not a list but a single element"`
` i am a list`
` "i;am;a;list"`

File: scripts/cmake/vcpkg_build_make.cmake [hunk 1]
```diff
@@ -52,10 +52,10 @@ function(vcpkg_build_make)
             set(ENV{PATH} "$ENV{PATH};${YASM_EXE_PATH};${MSYS_ROOT}/usr/bin;${PERL_EXE_PATH}")
             set(BASH ${MSYS_ROOT}/usr/bin/bash.exe)
             # Set make command and install command
-            set(MAKE ${BASH} --noprofile --norc -c "${_VCPKG_PROJECT_SUBPATH}make")
+            set(MAKE ${BASH} --noprofile --norc -c ${_VCPKG_PROJECT_SUBPATH}make)
```
Comment (Neumann-A) [inline scripts/cmake/vcpkg_build_make.cmake]: those quotes seem to be required to have paths with spaces

**Comment by Christsnatcher** [comment]
@Neumann-A: The removal of these particular quotes apparently isn't necessary, I just tried and x264 building worked flawlessly with the quotes in place.

File: scripts/cmake/vcpkg_build_make.cmake (hunk 1 above)
Comment (JackBoosY) [reply scripts/cmake/vcpkg_build_make.cmake]: @Neumann-A Adding double quotes here will break the integrity of `make` and `install`, causing the build to fail.

**Comment by oracleoftroy** [comment]
@JackBoosY I took a look at the quotes on line 55 this morning and I can confirm what @Christsnatcher says. I added quotes around just the make executable, changing line 55 to:

`set(MAKE ${BASH} --noprofile --norc -c "${_VCPKG_PROJECT_SUBPATH}make")`

I had no issue installing x264 with this change.

**Comment by Neumann-A** [comment]
@JackBoosY: Got this error in WSL with this PR merged into my libpq PR:
```
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
    Command failed: "make -j" 9
    Working Directory: /mnt/d/vcpkg_test/libpq_port/buildtrees/libpq/x64-linux-dbg
    See logs for more information:
```

**Comment by Neumann-A** [comment]
Solved it with `COMMAND "${MAKE};${MAKE_OPTS}"` but this will brake your windows path. The problem is you are building the command wrongly. You want to build the bash and make command separatly. You then need to call excute_build_process with a list of commands in relation to the bash command and have the make command be passed as a single string to the bash command.

**Comment by JackBoosY** [comment]
@Neumann-A I don't understand the difference between command and parameter splitting between windows and linux, but it is successful with your workaround.

Good work!

**Comment by timautry** [comment]
Thank you @JackBoosY  - does the job on both VS2017 and VS2019 latest builds. I appreciate the hard work you put in on this. Have a wonderful week.

**Comment by Christsnatcher** [comment]
Working perfectly now, a huge "Thank you" to everyone involved in fixing this issue!

**Pull request merged**